moveit_simple/include/moveit_simple/moveit_simple.h

```diff
@@ -0,0 +1,355 @@
+   * prior calls to "addTrajPoint".
+   * @return - true if desired trajectory was executed.
+   */
+  bool execute(const std::string traj_name);
+  /**
+   * @brief clearTrajectory - clears stored trajectory
+   * @param traj_name - trajectory to clear
+   */
+  void clearTrajectory(const::std::string traj_name);
+
+  /**
+   * @brief toJointTrajPtMsg - Converts native joint point (vector + time) to ROS joint
+   * trajectory point message type.
+   * @param joint_point
+   * @param time
+   * @return
+   */
+  static trajectory_msgs::JointTrajectoryPoint toJointTrajPtMsg(
+      const std::vector<double> & joint_point, double time);
+
+protected:
+  Robot();
+
+  Eigen::Affine3d transformToBase(const Eigen::Affine3d &in,
+                                         const std::string &in_frame) const;
+
+  bool toJointTrajectory(const std::string traj_name,
+                         std::vector<trajectory_msgs::JointTrajectoryPoint> & points);
+
+  bool addTrajPoint(const std::string & traj_name,
+                    std::unique_ptr<TrajectoryPoint> &point);
+
+  bool isReachable(std::unique_ptr<TrajectoryPoint> & point, double timeout,
+                   std::vector<double> joint_seed = std::vector<double>() ) const;
+
+
+  bool getIK(const Eigen::Affine3d pose, const std::vector<double> & seed,
+             std::vector<double> & joint_point, double timeout=1,
+              unsigned int attempts=1) const;
+  bool getIK(const Eigen::Affine3d pose, std::vector<double> & joint_point,
+             double timeout=1, unsigned int attempts=1) const;
+
+  std::unique_ptr<TrajectoryPoint> lookupTrajectoryPoint(const std::string & name,
+                                                              double time) const;
+
+  bool isConfigChange(const std::vector<double> jp1,
+                      const std::vector<double> jp2) const;
+
+  trajectory_msgs::JointTrajectoryPoint toJointTrajPtMsg(
+      const JointTrajectoryPoint & joint_point) const;
+
+  void updateState(const sensor_msgs::JointStateConstPtr& msg);
+
+
+  // Robot internal objects
+  std::map<std::string, Trajectory> traj_map_;
+
+  // MoveIt objects
+  mutable moveit::core::RobotStatePtr robot_state_;
+  planning_scene::PlanningScenePtr planning_scene_;
+  const moveit::core::JointModelGroup *joint_group_;
+  robot_model::RobotModelConstPtr robot_model_ptr_;
+  robot_model_loader::RobotModelLoaderPtr robot_model_loader_;
+
+  // Visualizations
+  moveit_visual_tools::MoveItVisualToolsPtr visual_tools_;
+
+  // Move robot action
+  actionlib::SimpleActionClient<control_msgs::FollowJointTrajectoryAction> action_;
+
+  // TF for looking up waypoints
+  // TODO: Consider swapping out the listener for an action based lookup - much
+  // less overhead
+  tf2_ros::Buffer tf_buffer_;
+  tf2_ros::TransformListener tf_listener_;
+
+  // ROS objects
+  ros::NodeHandle nh_;
+  ros::Subscriber j_state_sub_;
+  mutable std::recursive_mutex m_;
+
+};
+
+
+
+
+class TrajectoryPoint
```

Comment:
No.  These are lightweight, and my intent is to hide them completely from the user.  The interface (public methods) doesn't use them.  In the future we are free to use whatever classes we want, likely Descartes.  However, at this point, I didn't want to create a dependency on Descartes.